File: a-string-repeating-function.py
def repeat_string_num_times(str_val,num):
    if num <= 0:
        return ""
    result = ""
    for _ in range(num):
        result += str_val
    return result

#variant

def repeat_string_num_times_smart(str_val,num):
    return str_val * num if num > 0 else ""

File: a-string-repeating-function.js
function repeatStringNumTimes(str, num) {
  if (num <= 0) {
    return "";
  }

  let result = "";

  for (let i = 0; i < num; i++) {
    result += str; 
  }

  return result;
}

//variant

function repeatStringNumTimes(str,num) {
    let result = "";
    while(num > 0) {
        result += str;
        num--;
    }
    return result;
}

//variant 2

function repeatStringNumTimes(str,num) {
    if (num <= 0) {
        return "";
    }
    return str + repeatStringNumTimes(str,num -1)
}

//for {start; goal; skip};